File: system-test/java/src/main/java/com/optivem/eshop/dsl/core/usecase/external/clock/usecases/base/BaseClockUseCase.java
package com.optivem.eshop.dsl.core.usecase.external.clock.usecases.base;

import com.optivem.eshop.dsl.driver.port.external.clock.ClockDriver;
import com.optivem.eshop.dsl.core.shared.BaseUseCase;
import com.optivem.eshop.dsl.core.shared.UseCaseContext;

public abstract class BaseClockUseCase<TSuccessResponse, TSuccessVerification> extends BaseUseCase<ClockDriver, TSuccessResponse, TSuccessVerification> {
    protected BaseClockUseCase(ClockDriver driver, UseCaseContext context) {
        super(driver, context);
    }
}

File: system-test/typescript/src/testkit/dsl/core/usecase/external/clock/usecases/base/BaseClockUseCase.ts
import type { ClockDriver } from '../../../../../../../driver/port/external/clock/clock-driver.js';
import { BaseUseCase } from '../../../../../shared/base-use-case.js';

export abstract class BaseClockUseCase<TResponse, TVerification> extends BaseUseCase<
  ClockDriver,
  TResponse,
  TVerification
> {}
